tests/results.cpp

```diff
@@ -1042,7 +1042,14 @@ TEST_CASE("notifications: sync") {
         // Start the server and wait for the Realm to be uploaded so that sync
         // makes some writes to the Realm and bumps the version
         server.start();
-        SyncManager::shared().get_session(config.path, *config.sync_config)->wait_for_upload_completion_blocking();
+        std::condition_variable cv;
+        std::mutex wait_mutex;
+        SyncManager::shared().get_session(config.path, *config.sync_config)->wait_for_upload_completion([&](auto) {
+            cv.notify_one();
+        });
+        std::unique_lock<std::mutex> lock(wait_mutex);
+        cv.wait(lock);
```

Comment:
`std::condition_variable::wait` can return spuriously, so there's the potential for this code to continue executing prior to the upload completing. I think you you can use the variant that takes a predicate, combined with the upload completion handler setting a flag, to avoid problems due to spurious unblocking.